src/mapwindow.jl

```diff
@@ -72,14 +73,18 @@ function mapwindow(f, img, window; border="replicate",
               resolve_imginds(indices))
 end
 
+# a unit range `r` having `r == axes(r)` which keeps its axes with
+# broadcasting (e.g., `axes(r.+1) == axes(r)`), unlike Base.IdentityUnitRange
+self_offset(r::AbstractUnitRange) = OffsetArrays.IdOffsetRange(1:length(r), first(r)-one(eltype(r)))
```

Comment:
I'm not exactly sure I understand what you're trying to do. It doesn't quite do what it claims in the comment, for example
```julia
julia> r2 = OffsetArrays.IdOffsetRange(5:10, -3)
2:7

julia> r3 = self_offset(r2)
2:7

julia> first(r3)
2

julia> first(axes(r2, 1))
-2
```
What's the underlying goal? Would you like to create a range that preserves the values and indices of an original `AbstractUnitRange` but which behaves as described under broadcasting?

Or is this a situation where we can essentially ignore certain axes? For instance, we might imagine ignoring the axes of `imginds`, although alternatively they might specify the axes of the output.

Reply:
I suppose the comment is confusing because it uses the name `r` for the output of the function when it's also as the name of the function parameter. It should really say that `s = self_offset(r)` has the same values as `r` but has itself as indices: `collect(s) == collect(r)` and `axes(s) == (s,)` (in your example, `axes(r3) == (2:7,) == (r3,)`).

Reply:
Fixed the comment